components/support/rs_client/src/http_client.rs

```diff
@@ -76,14 +92,15 @@ impl Client {
     }
 
     fn ensure_no_backoff(&self) -> Result<()> {
+        let current_state = self.remote_state.lock().unwrap().clone();
```

Comment:
Can we get away without the clone here? if we remove the clone then simply deref `current_state` (by using `*current_state`) we might be able to avoid the clone!